lib/api/BackbeatAPI.js

```diff
@@ -501,6 +501,56 @@ class BackbeatAPI {
         });
     }
 
+    /**
+     * Find the matching metric timestamp key and decrement it accordingly.
+     * @param {Array} keys - Array of keys that matched the query string
+     * @param {Number} timestamp - The normalized timestamp of the failed key
+     * @param {Number} decrement - The value to decrement the matching key by
+     * @param {Function} cb - Callback to call
+     * @return {undefined}
+     */
+    _decrementIntervalKey(keys, timestamp, decrement, cb) {
+        if (keys.length === 0) {
+            return process.nextTick(cb);
+        }
+        const intervalKey = keys.find(key => {
+            const schema = key.split(':');
+            return timestamp === schema[schema.length - 1];
+        });
+        const cmds = [['decrby', intervalKey, decrement]];
+        return this._redis.batch(cmds, (err, res) => {
```

Comment:
`this._batch.decrby`, this should work.

Reply:
Same as above: this is the `RedisClient` wrapper from Arsenal, not `ioredis`.